iati/core/rulesets.py

```diff
@@ -271,18 +270,41 @@ def is_valid_for(self, dataset):
         """Assert that the date value of `less` is older than the date value of `more`.
 
         Args:
-            dataset.xml_tree: an etree created from an XML dataset.
+            dataset (iati.core.Dataset): An IATI Dataset object.
 
         Return:
-            A boolean value. If `less` is older than `more`, return `True`.
+            bool: Changes depending on whether `less` is older than `more`.
+
+        Raises:
+            AttributeError: When an argument is given that is not a dataset object.
+            ValueError: When a date is given that is not in the correct xsd:date format.
+            XPathEvalError: When no valid xpath argument is given.
+
+        Note:
+            `date` restricted to 10 characters in order to exclude possible timezone values but suboptimal solution.
```

Comment:
Because of this implementation detail, a string shorter than 10 characters should be tested.